src/main/scala/uk/gov/nationalarchives/dp/client/EntityClient.scala

```diff
@@ -704,63 +704,8 @@ object EntityClient {
       parentRef: Option[UUID]
   )
 
-  /** An object providing a method to return a SecurityTag instance from a string
-    */
-  object SecurityTag {
-
-    /** Returns a security tag from a string
-      * @param securityTagString
-      *   The security tag as a string. Either 'open' or 'closed'
-      * @return
-      *   The SecurityTag instance
-      */
-    def fromString(securityTagString: String): Option[SecurityTag] = securityTagString match {
-      case "open"   => Option(Open)
-      case "closed" => Option(Closed)
-      case _        => None
-    }
-  }
-
-  /** Represents an Open security tag
-    */
-  case object Open extends SecurityTag
-
-  /** Represents a closed security tag
-    */
-  case object Closed extends SecurityTag
-
-  /** A structural object
-    */
-  case object StructuralObject extends EntityType {
-    override val entityPath = "structural-objects"
-    override val entityTypeShort: String = "SO"
-  }
-
-  /** An information object
-    */
-  case object InformationObject extends EntityType {
-    override val entityPath = "information-objects"
-    override val entityTypeShort: String = "IO"
-  }
-
-  /** A content object
-    */
-  case object ContentObject extends EntityType {
-    override val entityPath = "content-objects"
-    override val entityTypeShort: String = "CO"
-  }
-
-  case object Access extends RepresentationType
-
-  case object Preservation extends RepresentationType
-
-  sealed trait GenerationType {
-    override def toString: String = getClass.getSimpleName.dropRight(1).toLowerCase
-  }
-
-  case object Original extends GenerationType
-
-  case object Derived extends GenerationType
+  enum GenerationType:
+    case Original, Derived
```

Comment:
Should we add a `toString` method for this since it was there before?

Reply:
You don't need it. There's a built in toString method which returns Original or Derived depending.

Reply:
For this one in particular, the previous `toString` method also lowercased the class name e.g.
```
 sealed trait GenerationType {
    override def toString: String = getClass.getSimpleName.dropRight(1).toLowerCase
  }
```

I think for the purpose of the disaster recovery; I guess we could just lowercase it in the disaster recovery code?

Reply:
Yeah, I think that's better to do it in the disaster recovery rather than overriding the default.